### PR TITLE
ci: fix dependabot script not bumping packages in changeset

### LINKED ISSUE #2594: Dependabot script not bumping patch versions in changesets
If a dependency in one of our packages is bumped, the [changesets](https://github.com/FuelLabs/fuels-ts/pull/2588/files#r1650213697) should also bump the patch version of that package.

### PR DESCRIPTION
**Pull request by nedsalk**
I verified the script works by:
1. Checking out #2588 branch to which the closing issue links
2. Checking out the commit on which the faulty script ran (95f9484e279e7fe270f954692a77a71ac3ef8cb1)
3. Running the script on that commit and verifying that it doesn't give the expected results
4. Running `git checkout ns/chore/dependabot-bump-patch scripts/changeset/dependabot-changeset.sh` to get this branch's changes
5. Rerunning the script and getting the expected results in the changeset file

Besides fixing the problem, I also added monitoring of `templates` changes and bumping `create-fuels` if a change happened there.

- closes #2594

### CODE REVIEW

File: scripts/changeset/dependabot-changeset.sh [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+# Get the first dependabot commit which contains the changed files
+first_commit=$(git log master..HEAD --oneline --reverse | xargs | awk '{print $1;}')
 
-# Get the list of changed package directories
-changed_packages=$(git diff-tree --no-commit-id --name-only -r "$latest_commit" | grep "packages/" | sed -E 's|packages/([^/]+)/.*|\1|')
+changed_files=$(git diff-tree --no-commit-id --name-only -r "$first_commit")
+
+# Get the list of relevant package.json changes
+changed_packages=$(echo "$changed_files" | grep -E "packages/.+/package.json" | xargs jq -r '.name')
+changed_templates=$(echo "$changed_files" | grep -E "templates/.+/package.json")
```
Comment (maschad) [inline scripts/changeset/dependabot-changeset.sh:10]: Let's log these out as well.

```suggestion
changed_packages=$(echo "$changed_files" | grep -E "packages/.+/package.json" | xargs jq -r '.name')
changed_templates=$(echo "$changed_files" | grep -E "templates/.+/package.json")

echo "changed packages: $changed_packages"
echo "changed templates: $changed_templates"
```

File: scripts/changeset/dependabot-changeset.sh [hunk 2]
```diff
@@ -1,17 +1,27 @@
 #!/bin/bash
 
-# Get the latest commit hash on the current branch
-latest_commit=$(git rev-parse HEAD)
+# Get the first dependabot commit which contains the changed files
+first_commit=$(git log master..HEAD --oneline --reverse | xargs | awk '{print $1;}')
 
-# Get the list of changed package directories
-changed_packages=$(git diff-tree --no-commit-id --name-only -r "$latest_commit" | grep "packages/" | sed -E 's|packages/([^/]+)/.*|\1|')
+changed_files=$(git diff-tree --no-commit-id --name-only -r "$first_commit")
```
Comment (maschad) [inline scripts/changeset/dependabot-changeset.sh:6]:
```suggestion
changed_files=$(git diff-tree --no-commit-id --name-only -r "$first_commit")
echo "changed files: $changed_files"
```

File: scripts/changeset/dependabot-changeset.sh [hunk 3]
```diff
@@ -1,17 +1,27 @@
 #!/bin/bash
 
-# Get the latest commit hash on the current branch
-latest_commit=$(git rev-parse HEAD)
+# Get the first dependabot commit which contains the changed files
+first_commit=$(git log master..HEAD --oneline --reverse | xargs | awk '{print $1;}')
```
Comment (maschad) [inline scripts/changeset/dependabot-changeset.sh:4]:
```suggestion
first_commit=$(git log master..HEAD --oneline --reverse | xargs | awk '{print $1;}')
echo "First commit: $first_commit"
```

**Review by maschad — APPROVED** [review]
This is great work. My only suggestion would be given this has failed before in the past, it may be useful to have some logging added should this have issues going forward.

**Review by maschad — DISMISSED** [review]

**Review by arboleya — APPROVED** [review]

File: scripts/changeset/dependabot-changeset.sh [hunk 4]
```diff
@@ -1,17 +1,33 @@
...
+echo "First commit: $first_commit"
 
-# Get the list of changed package directories
-changed_packages=$(git diff-tree --no-commit-id --name-only -r "$latest_commit" | grep "packages/" | sed -E 's|packages/([^/]+)/.*|\1|')
+changed_files=$(git diff-tree --no-commit-id --name-only -r "$first_commit")
+echo "changed files: $changed_files"
+
+# Get the list of relevant package.json changes
+changed_packages=$(echo "$changed_files" | grep -E "packages/.+/package.json" | xargs jq -r '.name')
+changed_packages=$(echo "$changed_files" | grep -E "packages/.+/package.json" | xargs jq -r '.name')
```
Comment (arboleya) [inline scripts/changeset/dependabot-changeset.sh:12]:
```suggestion
# Get the list of relevant package.json changes
changed_packages=$(echo "$changed_files" | grep -E "packages/.+/package.json" | xargs jq -r '.name')
```

**Review by arboleya — CHANGES_REQUESTED** [review]

**Review by maschad — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Pull request merged**